client/src/main/scala/com.gu.contentapi.client/ScheduledExecutor.scala

```diff
@@ -0,0 +1,48 @@
+package com.gu.contentapi.client
+
+import java.util.concurrent.ThreadPoolExecutor.AbortPolicy
+import java.util.concurrent.{
+  Executors,
+  ScheduledExecutorService,
+  RejectedExecutionHandler,
+  RejectedExecutionException
+}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+import scala.concurrent.Promise
+import scala.language.implicitConversions
+
+object ScheduledExecutor {
+  private val defaultHandler: RejectedExecutionHandler = new AbortPolicy
+}
+
+/**
+  * A single threaded executor for tasks scheduled in the future
+  * @throws IllegalArgumentException if { @code corePoolSize < 0}
+  * @throws NullPointerException if { @code threadFactory} or
+  *                                         { @code handler} is null
+  */
+class ScheduledExecutor {
+
+  private lazy val underlying: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
```

Comment:
This is an implementation choice, but the user of the library may have its own scheduler they prefer to use, if they are resource-conscious. So here is what I would do:

1. make the class abstract with only `sleepFor`
2. define a default implementation (this one) in the companion object, make it private and add `ScheduledExecutor.apply` that creates an instance of it

With that, users can then create their own version of `ScheduledExecutor` if they want to.